src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,146 @@
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+
+  const walletConnectIsActive = walletConnectHooks.useIsActive()
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const fortmaticIsActive = fortmaticHooks.useIsActive()
+  const previousFortmaticIsActive = usePrevious(fortmaticIsActive)
+
+  const [isEagerlyConnecting, setIsEagerlyConnecting] = useState(false)
+
+  // The dependency list is empty so this is only run once on mount
+  useEffect(() => {
+    connect(gnosisSafe)
+    connect(network)
+
+    if (walletOverride) {
+      connect(getConnectorForWallet(walletOverride))
+      setIsEagerlyConnecting(true)
+    } else if (!walletOverrideBackfilled) {
+      MODAL_WALLETS.filter((wallet) => wallet !== Wallet.FORTMATIC)
+        .map(getConnectorForWallet)
+        .forEach(connect)
+      setIsEagerlyConnecting(true)
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    const injectedState: ConnectorState = {
+      isActive: injectedIsActive,
+      previousIsActive: previousInjectedIsActive,
+    }
+    const coinbaseWalletState: ConnectorState = {
+      isActive: coinbaseWalletIsActive,
+      previousIsActive: previousCoinbaseWalletIsActive,
+    }
+    const walletConnectState: ConnectorState = {
+      isActive: walletConnectIsActive,
+      previousIsActive: previousWalletConnectIsActive,
+    }
+    const fortmaticState: ConnectorState = {
+      isActive: fortmaticIsActive,
+      previousIsActive: previousFortmaticIsActive,
+    }
+    const isActiveMap = new Map<Wallet, ConnectorState>([
+      [Wallet.INJECTED, injectedState],
+      [Wallet.COINBASE_WALLET, coinbaseWalletState],
+      [Wallet.WALLET_CONNECT, walletConnectState],
+      [Wallet.FORTMATIC, fortmaticState],
+    ])
+
+    isActiveMap.forEach((state: ConnectorState, wallet: Wallet) => {
+      const { isActive, previousIsActive } = state
+      if (isActive && !previousIsActive) {
+        // When a user manually sets their new connection, set a wallet override.
+        // Also set an override when they were a user prior to this state being introduced.
+        if (!isEagerlyConnecting || !walletOverrideBackfilled) {
+          dispatch(updateWalletOverride({ wallet }))
+        }
+
+        // Reset the eagerly connecting state.
+        if (isEagerlyConnecting) {
+          setIsEagerlyConnecting(false)
+        }
+      }
+    })
+  }, [
+    dispatch,
+    walletOverride,
+    walletOverrideBackfilled,
+    injectedIsActive,
+    coinbaseWalletIsActive,
+    walletConnectIsActive,
+    previousInjectedIsActive,
+    previousCoinbaseWalletIsActive,
+    previousWalletConnectIsActive,
+    fortmaticIsActive,
+    previousFortmaticIsActive,
+    isEagerlyConnecting,
+    setIsEagerlyConnecting,
+  ])
+
+  return null
+}
+
+interface Props {
+  children: JSX.Element
+}
+
+export default function Web3Provider({ children }: Props) {
```

Comment:
nit: if just children, then just inline the props: `{ children }: { children: ReactNode }` (with `import { ReactNode } from 'react'`